retrodep/gosource_test.go

```diff
@@ -310,3 +311,82 @@ func TestImportPathFromFilepath(t *testing.T) {
 		}
 	}
 }
+
+type mockWorkingTree struct{ stubWorkingTree }
+
+func (m *mockWorkingTree) FileHashesFromRef(ref, subPath string) (FileHashes, error) {
+	hashes := make(FileHashes)
+	// This is the correct hash for nl.go:
+	hashes["nl.go"] = "4ccdb7b17d6eaf1b51ed56932c020edcf323fd5734ce32d01a2713edeb17f6da"
+	return hashes, nil
+}
+
+func TestGoSourceDiff(t *testing.T) {
+	dir := "testdata/godep"
+	src, err := NewGoSource(dir, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Reset vcsRepoRootForImportPath after this test.
+	defer func() {
+		vcsRepoRootForImportPath = vcs.RepoRootForImportPath
+	}()
+
+	vcsRepoRootForImportPath = func(importPath string, _ bool) (*vcs.RepoRoot, error) {
+		return &vcs.RepoRoot{
+			Root: "example.com/foo/bar",
+		}, nil
+	}
+
+	project, err := src.Project("example.com/foo/bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wt := &mockWorkingTree{
+		stubWorkingTree: stubWorkingTree{
+			anyWorkingTree: anyWorkingTree{
+				hasher: &sha256Hasher{},
+			},
+		},
+	}
+
+	writer := &strings.Builder{}
+	changes, err := src.Diff(project, wt, writer, dir, "v1.0.0")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if changes != true {
+		t.Errorf("changes: got %t but expected %t", changes, true)
+	}
+
+	// Find which files are in the diff output.
+	output := writer.String()
+	lines := strings.Split(output, "\n")
+	new := make(map[string]struct{})
```

Comment:
new is a builtin function, better use newSomething.

Reply:
Thanks, will change.